test/jasmine/tests/plot_api_react_test.js

```diff
@@ -1806,3 +1807,67 @@ describe('Plotly.react and uirevision attributes', function() {
         _run(fig, editComponents, checkInitial, checkEdited).then(done);
     });
 });
+
+describe('Test Plotly.react + interactions under uirevision:', function() {
+    var gd;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(function() {
+        Plotly.purge(gd);
+        destroyGraphDiv();
+    });
+
+    it('@gl gl3d subplots preserve camera changes on interactions', function(done) {
+        function _react() {
+            return Plotly.react(gd, [{
+                type: 'surface',
+                z: [[1, 2, 3], [3, 1, 2], [2, 3, 1]]
+            }], {
+                width: 500,
+                height: 500,
+                uirevision: true
+            });
+        }
+
+        // mocking panning/scrolling is brittle,
+        // this here is enough to to trigger the relayoutCallback
```

Comment:
but this presumably doesn't actually change `camera.eye`, right? Would it be possible to do something that changes the `eye`, even if it's not robust what it changes it *to*? Just "not the default", and verify that after another `_react` it's again "not the default"? Or is it so brittle we can't even ensure it will register anything at all?

Reply:
> but this presumably doesn't actually change `camera.eye`, right?

Yes, you were correct. 

> Would it be possible to do something that changes the `eye`, even if it's not robust what it changes it _to_? Just "not the default", and verify that after another `_react` it's again "not the default"?

Good call. Here's my attempt -> https://github.com/plotly/plotly.js/pull/3394/commits/b880717ba8b86602a62d8010a8028695556c0103

Reply:
Ah clever! I'll take it :)